### PR TITLE
K8s Support for SDK

### PR DESCRIPTION
**Pull request by Maxusmusti**
Switching from OpenShift client to Kubernetes Python API

For issue #139

### CODE REVIEW

File: src/codeflare_sdk/cluster/cluster.py [hunk 1]
```diff
@@ -243,16 +245,21 @@ def cluster_dashboard_uri(self) -> str:
...
-                )
-                route = route.out().split(" ")
-                route = [x for x in route if f"ray-dashboard-{self.config.name}" in x]
-                route = route[0].strip().strip("'")
-            return f"http://{route}"
-        except:
-            return "Dashboard route not available yet, have you run cluster.up()?"
+            config.load_kube_config()
+            api_instance = client.CustomObjectsApi()
+            routes = api_instance.list_namespaced_custom_object(
```
Comment (astefanutti) [inline src/codeflare_sdk/cluster/cluster.py:250]: Ideally, to support running the SDK in-cluster, the dashboard URI would resolves to the dashboard service endpoint, and would not require Ingress / Route.

File: src/codeflare_sdk/job/jobs.py [hunk 2]
```diff
@@ -17,13 +17,13 @@
...
 from pathlib import Path
 
-import openshift as oc
 from torchx.components.dist import ddp
 from torchx.runner import get_runner
 from torchx.specs import AppHandle, parse_app_handle, AppDryRunInfo
 
 if TYPE_CHECKING:
     from ..cluster.cluster import Cluster
+from ..cluster.cluster import get_current_namespace
```
[the remark this answers is not in the document]
Comment (Maxusmusti) [reply src/codeflare_sdk/job/jobs.py:26]: Potentially, though I think that'd be best as a separate issue/PR (reorganization of our global functions that are scattered across modules)

File: src/codeflare_sdk/cluster/cluster.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Maxusmusti) [reply src/codeflare_sdk/cluster/cluster.py:255]: We could change the SDK code for `cluster.cluster_dashboard_uri()` to get the ingress domain and provide the correct uri without using the route API, but I believe that will not solve the underlying problem of how the route is being created in the first place:
```
    - generictemplate:
        apiVersion: route.openshift.io/v1
        kind: Route
        metadata:
          labels:
            odh-ray-cluster-service: quicktest-head-svc
          name: ray-dashboard-quicktest
          namespace: default
        spec:
          port:
            targetPort: dashboard
          to:
            kind: Service
            name: quicktest-head-svc
      replica: 1
    Items: []
```
which I believe will not work currently in Kubernetes anyways. So tracking this as a separate conversation is likely the way to go for now.

[the remark this answers is not in the document]
Comment (Maxusmusti) [reply src/codeflare_sdk/cluster/cluster.py:255]: Ah, yeah just noticed there is a separate issue to track that effort: https://github.com/project-codeflare/codeflare-sdk/issues/219

I will add a note to that issue

[the remark this answers is not in the document]
Comment (MichaelClifford) [reply src/codeflare_sdk/cluster/cluster.py:255]: Yeah, we only support OCP routes at the moment, but happy to see with have an issue for ingress

**Review by MichaelClifford — APPROVED** [review]
LGTM!

**Review by carsonmh — APPROVED** [review]
LGTM!

**Pull request merged**